src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -2413,6 +2413,51 @@ internal static string ResolveFilePath(IScriptExtent errorExtent, object obj, Ex
                 throw;
             }
         }
+
+        internal static string ResolveLiteralFilePath(IScriptExtent errorExtent, object obj, ExecutionContext context)
+        {
+            try
+            {
+                FileInfo file = obj as FileInfo;
+                string filePath = file != null ? file.FullName : PSObject.ToStringParser(context, obj);
+
+                if (string.IsNullOrEmpty(filePath))
+                {
+                    throw InterpreterError.NewInterpreterException(filePath,
```

Comment:
We could remove this because below we get ItemNotFoundException from GetUnresolvedProviderPathFromPSPath()